### PR TITLE
fix network economics API route

### PR DESCRIPTION
**Pull request by miiu96**
Bugfix: change mechanism how the `total-staked` value and the `top-up` value is calculated

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by BeniaminDrasovean — DISMISSED** [review]

File: node/node_test.go [hunk 1]
```diff
@@ -3021,3 +3038,84 @@ func TestNode_StartHeartbeat(t *testing.T) {
 
 	time.Sleep(time.Second)
 }
+
+func TestGetKeyValuePairs_CannotDecodeAddress(t *testing.T) {
```
Comment (iulianpascalau) [inline node/node_test.go:3042]: :+1:

File: node/stakeValuesProcessor/stakeValuesProcessor.go [hunk 2]
```diff
@@ -2,46 +2,55 @@ package stakeValuesProcessor
...
 
 type stakedValuesProc struct {
-	marshalizer marshal.Marshalizer
-	accounts    state.AccountsAdapter
-	nodePrice   *big.Int
+	accounts           state.AccountsAdapter
+	blockChain         data.ChainHandler
+	queryService       process.SCQueryService
+	publicKeyConverter core.PubkeyConverter
+	mutex              *sync.Mutex
```
Comment (iulianpascalau) [inline node/stakeValuesProcessor/stakeValuesProcessor.go:25]: could have renamed this to `mutexComputeStakedValueAndTopUp`

File: node/stakeValuesProcessor/stakeValuesProcessor.go [hunk 3]
```diff
@@ -80,29 +102,47 @@ func (svp *stakedValuesProc) computeStakedValueAndTopUp() (*big.Int, *big.Int, e
...
 
-	numRegistedNodes := uint64(0)
-	totalStakedValueAllLeaves := big.NewInt(0)
+	totalStaked, totalTopUp := big.NewInt(0), big.NewInt(0)
 	for leaf := range chLeaves {
-		validatorData := &systemSmartContracts.ValidatorDataV2{}
-		value, errTrim := leaf.ValueWithoutSuffix(append(leaf.Key(), vm.ValidatorSCAddress...))
-		if errTrim != nil {
-			return nil, nil, fmt.Errorf("%w for validator key %s", errTrim, hex.EncodeToString(leaf.Key()))
+		if len(leaf.Key()) != svp.publicKeyConverter.Len() {
```
Comment (iulianpascalau) [inline node/stakeValuesProcessor/stakeValuesProcessor.go:107]: Slight optimization: could have "cached" the `svp.publicKeyConverter.Len()` in a variable on L105.

**Review by iulianpascalau — APPROVED** [review]

File: node/stakeValuesProcessor/stakeValuesProcessor.go (hunk 3 above)
Comment (miiu96) [reply node/stakeValuesProcessor/stakeValuesProcessor.go:107]: done.

**Review by iulianpascalau — APPROVED** [review]

File: node/stakeValuesProcessor/stakeValuesProcessor.go (hunk 3 above)
Comment (iulianpascalau) [reply node/stakeValuesProcessor/stakeValuesProcessor.go]: fair enough. You cached the leafKey :D

**Review by sasurobert — APPROVED** [review]

File: node/stakeValuesProcessor/stakeValuesProcessor.go (hunk 3 above)
Comment (miiu96) [reply node/stakeValuesProcessor/stakeValuesProcessor.go]: :+1:

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Pull request merged**